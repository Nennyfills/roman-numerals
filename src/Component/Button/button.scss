.button {
  button {
    width: auto;
    display: inline-block;
    margin-right: 0.5rem;
    margin-bottom: 0.8rem;
    padding: 8px 16px;
    border-radius: 5px;
    text-align: center;
    border: none;
    cursor: pointer;
    font-weight: 600;
    letter-spacing: -0.006px;
    height: 42px;
    font-size: 16px;
    line-height: 8px;
    background-color: #3ab795;
    color: #ffffff;
    outline: none;
    &:disabled{
      color: #ffff;
      box-shadow: none;
      cursor: not-allowed;
      background: #b0e4d6;
      border-radius: 6px;
      &:hover {
        text-decoration: none;
        background: #b0e4d6;
        box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.1);
        border-color: #b0e4d6;
      }
    }
    &:hover {
      text-decoration: none;
      background: #2a9d89;
      box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.1);
      border-color: #2a9d89;
    }
  }
}
