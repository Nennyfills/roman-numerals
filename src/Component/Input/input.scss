.input {
  input,
  label {
    display: block;
    font-weight: 500;
    font-size: 16px;
    line-height: 8px;
    padding-bottom: 8px;
    color: #222222;
    text-align: left;
  }
  input {
    width: 100%;
    display: inline-block;
    text-transform: uppercase;
    margin: 0.6rem 0.5rem 0.8rem 0;
    border-radius: 5px;
    border: 1px solid #dfe2e6;
    font-style: normal;
    padding: 12px;
    font-weight: normal;
    text-align: left;
    &::-webkit-outer-spin-button,
    &::-webkit-inner-spin-button {
      -webkit-appearance: none;
    }
    &[type='number'] {
      -moz-appearance: textfield;
    }

    &:focus {
      outline: none;
    }
  }
}
