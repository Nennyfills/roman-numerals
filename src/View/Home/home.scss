.home {
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
  .home_solution {
    border: 1px solid #3ab795;
    min-height: 1rem;
    width: 20rem;
    text-transform: uppercase;
    .solution_top, .solution_down {
      display: flex;
      justify-content: center;
      align-items: center;

      .top, .down {
        list-style-type: none;
        padding: 0 1rem;
      }
      .top_line{
        border-left: solid 2px #3ab795;
        height:1rem;
        margin-left: .3rem;
      }
      .down{
        margin-bottom: 1rem;
      }
    }
  }
  .home_action {
    .action_number,
    .action_numerals {
      width: 25rem;
      margin-top: 1rem;
      display: flex;
      justify-content: space-around;
      align-items: flex-end;
    }
  }
}
