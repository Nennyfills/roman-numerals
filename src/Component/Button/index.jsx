import React from 'react';
import PropTypes from 'prop-types';

import './button.scss';

const Button = ({
  name, onClick, disabled, btnId
}) => (
  <div className="button">
    <button
      type="submit"
      id={btnId}
      data-testid={btnId}
      onClick={onClick}
      disabled={disabled}
    >
      {name}
    </button>
  </div>
);

Button.defaultProps = {
  name: 'name',
  onClick: () => 'click',
  disabled: true,
  btnId: 'text',
};
Button.propTypes = {
  name: PropTypes.string,
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  btnId: PropTypes.string,
};
export default Button;
