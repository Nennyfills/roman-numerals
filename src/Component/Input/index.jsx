import React from 'react';
import PropTypes from 'prop-types';

import './input.scss';

const Input = ({
  type,
  label,
  placeholder,
  handleChange,
  name,
  value,
  inputId,
}) => (
  <div className="input">
    <label htmlFor={inputId}>{label}</label>
    <input
      data-testid={inputId}
      id={inputId}
      name={name}
      value={value}
      type={type}
      placeholder={placeholder}
      onChange={handleChange}
      autoComplete="off"
    />
  </div>
);

Input.defaultProps = {
  label: '',
  placeholder: '',
  name: '',
  value: '',
  handleChange: () => 'handleChange',

  inputId: 'test',
};
Input.propTypes = {
  type: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  label: PropTypes.string,
  placeholder: PropTypes.string,
  handleChange: PropTypes.func,
  name: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  inputId: PropTypes.string,
};
export default Input;
