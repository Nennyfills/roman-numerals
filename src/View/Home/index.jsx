import React, { useState } from 'react';

import Box from '../../Component/Box';
import Button from '../../Component/Button';
import Input from '../../Component/Input';
import { RomanNumerals } from '../../Helpers';
import './home.scss';

const Home = () => {
  const [letter, setLetter] = useState('');
  const [number, setNumber] = useState('');

  const handleChange = ({ target }) => {
    const changeValue = target.name === 'number'
      ? setNumber(target.value)
      : setLetter(target.value);
    return changeValue;
  };

  const handleLetterConvert = () => {
    const result = RomanNumerals.fromRoman(letter);
    setNumber(result);
  };
  const handleNumberConvert = () => {
    const result = RomanNumerals.toRoman(+number);
    setLetter(result);
  };

  return (
    <div className="home">
      <Box>
        <div className="home_action">
          <div className="action_numerals">
            <Input
              inputId="letter-convert"
              type="text"
              name="letter"
              label="Enter a valid roman numerals"
              value={letter}
              placeholder=""
              handleChange={handleChange}
            />
            <Button
              btnId="letter-convert-btn"
              onClick={handleLetterConvert}
              disabled={!letter}
              name="Convert"
            />
          </div>
          <div className="action_number">
            <Input
              inputId="number-convert"
              type="number"
              name="number"
              label="Enter a number"
              value={number}
              placeholder=""
              handleChange={handleChange}
            />
            <Button
              btnId="number-convert-btn"
              onClick={handleNumberConvert}
              disabled={!number}
              name="Convert"
            />
          </div>
        </div>
      </Box>
    </div>
  );
};

export default Home;
