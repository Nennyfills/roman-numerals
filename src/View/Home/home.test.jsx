import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';

import Home from './index';
import Input from '../../Component/Input';
import Button from '../../Component/Button';

describe('Home Page', () => {
  afterEach(cleanup);

  test('renders properly', () => {
    const { container } = render(<Home />);
    expect(container.firstChild).toMatchSnapshot(`
      <div />
    `);
  });

  it('should call the handleChange function and change the state with input text', () => {
    const { getByTestId } = render(
      <Home>
        <Input type="text" />
      </Home>
    );

    const inputLetter = getByTestId('letter-convert');

    fireEvent.change(inputLetter, { target: { value: 'xxx' } });

    expect(inputLetter.value).toBe('xxx');
  });
  it('should call the handleChange function and change the state with input number', () => {
    const { getByTestId } = render(
      <Home>
        <Input type="number" />
      </Home>
    );

    const inputNumber = getByTestId('number-convert');
    fireEvent.change(inputNumber, { target: { value: '30' } });
    expect(inputNumber.value).toBe('30');
  });

  it('test if state is changed when button clicked with input text', () => {
    const { getByTestId } = render(
      <Home>
        <Button />
        <Input type="number" />
      </Home>
    );
    const inputLetter = getByTestId('letter-convert');
    const btn = getByTestId('letter-convert-btn');

    fireEvent.change(inputLetter, { target: { value: 'xxx' } });
    fireEvent.click(btn);
    expect(getByTestId('number-convert').value).toBe('30');
  });

  it('test if state is changed when button clicked with input number', () => {
    const { getByTestId } = render(
      <Home>
        <Button />
        <Input type="text" />
      </Home>
    );
    const inputNumber = getByTestId('number-convert');
    const btn = getByTestId('number-convert-btn');

    fireEvent.change(inputNumber, { target: { value: 300 } });
    fireEvent.click(btn);
    expect(getByTestId('letter-convert').value).toBe('ccc');
  });
});
