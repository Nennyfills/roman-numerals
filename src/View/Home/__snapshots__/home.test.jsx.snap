// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page renders properly: 
      <div />
     1`] = `
<div
  class="home"
>
  <div
    class="box "
  >
    <div
      class="home_action"
    >
      <div
        class="action_numerals"
      >
        <div
          class="input"
        >
          <label
            for="letter-convert"
          >
            Enter a valid roman numerals
          </label>
          <input
            autocomplete="off"
            data-testid="letter-convert"
            id="letter-convert"
            name="letter"
            placeholder=""
            type="text"
            value=""
          />
        </div>
        <div
          class="button"
        >
          <button
            data-testid="letter-convert-btn"
            disabled=""
            id="letter-convert-btn"
            type="submit"
          >
            Convert
          </button>
        </div>
      </div>
      <div
        class="action_number"
      >
        <div
          class="input"
        >
          <label
            for="number-convert"
          >
            Enter a number
          </label>
          <input
            autocomplete="off"
            data-testid="number-convert"
            id="number-convert"
            name="number"
            placeholder=""
            type="number"
            value=""
          />
        </div>
        <div
          class="button"
        >
          <button
            data-testid="number-convert-btn"
            disabled=""
            id="number-convert-btn"
            type="submit"
          >
            Convert
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
